extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/webjar/WebJarUtil.java

```diff
@@ -0,0 +1,273 @@
+                @Override
+                public FileVisitResult preVisitDirectory(final Path dir,
+                        final BasicFileAttributes attrs) throws IOException {
+                    visitor.visitDirectory(rootFolderToCopy.relativize(dir).toString());
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFile(final Path file,
+                        final BasicFileAttributes attrs) throws IOException {
+                    String fileName = rootFolderToCopy.relativize(file).toString();
+
+                    String moduleName = getModuleOverrideName(resourcesArtifact, fileName);
+                    boolean overrideFileCreated = false;
+                    if (OVERRIDABLE_RESOURCES.contains(fileName)) {
+                        try (InsertVariableResult insertVariableResult = insertVariablesWithResult(userApplication, config,
+                                getOverride(userApplication.getResolvedPaths(), classLoader,
+                                        fileName, moduleName, webJar.getUseDefaultQuarkusBranding()),
+                                fileName)) {
+                            if (insertVariableResult.inputStream != null) {
+                                overrideFileCreated = true;
+                                // Override (either developer supplied or Quarkus)
+                                visitor.filterAndVisitFile(fileName, insertVariableResult.inputStream, webJar.getFilter());
+                            }
+                        }
+                    }
+
+                    if (!overrideFileCreated) {
+                        try (InsertVariableResult insertVariableResult = insertVariablesWithResult(userApplication, config,
+                                Files.newInputStream(file), fileName)) {
+                            if (!webJar.getOnlyCopyNonArtifactFiles() || insertVariableResult.changed) {
+                                visitor.filterAndVisitFile(fileName, insertVariableResult.inputStream, webJar.getFilter());
+                            }
+                        }
+                    }
+
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+
+            if (fs != null) {
+                fs.close();
+            }
+        }
+    }
+
+    static ResolvedDependency getAppArtifact(CurateOutcomeBuildItem curateOutcomeBuildItem, ArtifactKey artifactKey) {
+        return getAppArtifact(curateOutcomeBuildItem, artifactKey.getGroupId(), artifactKey.getArtifactId());
+    }
+
+    static ResolvedDependency getAppArtifact(CurateOutcomeBuildItem curateOutcomeBuildItem, String groupId,
+            String artifactId) {
+        for (ResolvedDependency dep : curateOutcomeBuildItem.getApplicationModel().getDependencies()) {
+            if (dep.getArtifactId().equals(artifactId)
+                    && dep.getGroupId().equals(groupId)) {
```

Comment:
Keep in mind, it may happen there could be multiple artifacts on the classpath with the same G:A but different classifiers.

Reply:
We now also compare the classifier in this place.
I saw no other way to compare a GACT to the GACTV of the ResolvedDependency.